src/ConfigFileParser.cpp

```diff
@@ -0,0 +1,534 @@
+ *
+ * The copy constructor is private to prevent direct object creation.
+ * @param ref The ConfigFileParser object to copy
+ */
+ConfigFileParser::ConfigFileParser(const ConfigFileParser& ref)
+{
+    static_cast<void>(ref);
+}
+
+/**
+ * @brief PRIVATE Overload of the assignment operator.
+ *
+ * The assigment operator is private to prevent direct object creation.
+ * @param ref The ConfigFileParser object to assign.
+ * @return ConfigFileParser&  The ConfigFileParser object.
+ */
+ConfigFileParser& ConfigFileParser::operator=(const ConfigFileParser& ref)
+{
+    static_cast<void>(ref);
+    return *this;
+}
+
+/**
+ * @brief Parses the config file into a ConfigFile object and returns this object
+ *
+ * The function checks also if the config file:
+ * 1. Can be opened
+ * 2. Is not empty
+ * 3. Does not contain open brackets
+ * 4. Starts with 'http {'
+ * 5. Contains minimum one server
+ *
+ * @param configFilePath Path to the config file
+ * @return const ConfigFile& Created ConfigFile object
+ */
+const ConfigFile& ConfigFileParser::parseConfigFile(const std::string& configFilePath)
+{
+    m_configFile.stream.open(configFilePath.c_str());
+    if (!m_configFile.stream.is_open())
+        throw std::runtime_error("Failed to open config file");
+    if (m_configFile.stream.peek() == std::ifstream::traits_type::eof())
+        throw std::runtime_error("Config file is empty");
+
+    if (isBracketOpen(configFilePath))
+        throw std::runtime_error("Open bracket(s) in config file");
+
+    readAndTrimLine();
+    if (m_configFile.currentLine != "http {")
+        throw std::runtime_error("Config file does not start with 'http {'");
+
+    m_configFile.serverIndex = 0;
+    for (readAndTrimLine(); m_configFile.currentLine != "}"; readAndTrimLine()) {
+        if (m_configFile.currentLine == "server {") {
+            for (readAndTrimLine(); m_configFile.currentLine != "}"; readAndTrimLine())
+                readServerConfigLine();
+            m_configFile.serverIndex++;
+        }
+    }
+    if (m_configFile.servers.empty())
+        throw std::runtime_error("No server(s) in config file");
+
+    return m_configFile;
+}
+
+/**
+ * @brief Initializes the ConfigServer object with default values
+ *
+ * @param configServer The ConfigServer object to initialize
+ */
+void ConfigFileParser::initializeConfigServer(ConfigServer& configServer)
+{
+    const char* validServerDirectiveNames[] = { "server_name", "listen", "host", "client_max_body_size", "error_page", "location", "root" };
+    const int validServerDirectiveNamesSize = sizeof(validServerDirectiveNames) / sizeof(validServerDirectiveNames[0]);
+
+    configServer.validServerDirectives = std::vector<std::string>(validServerDirectiveNames, validServerDirectiveNames + validServerDirectiveNamesSize);
+    configServer.serverName = "";
+    configServer.root = "html";
+    configServer.listen.insert(std::make_pair("127.0.0.1", 80));
```

Comment:
As mentioned above, the init of the vars is now done in the constructor of the struct ConfigServer. 

The data type of the var for storing the port number is now a string instead of an unsigned short.